ast/term.go

```diff
@@ -2063,6 +2028,42 @@ func (obj *object) insert(k, v *Term) {
 	}
 }
 
+func (obj *object) remove(k *Term) {
+	hash := k.Hash()
+	head := obj.elems[hash]
+	equal := keyEqual(k)
+
+	var prev *objectElem = nil
```

Comment:
[nit] I think that's the same as 
```suggestion
	var prev *objectElem
```
but I guess it can't hurt to be explicit